types/protobuf.go

```diff
@@ -105,6 +105,7 @@ func (tm2pb) ConsensusParams(params *cmtproto.ConsensusParams) *abci.ConsensusPa
 		},
 		Evidence:  &params.Evidence,
 		Validator: &params.Validator,
+		Version:   &params.Version,
```

Comment:
I have a few non-blocking questions.
1. Is this function a translation layer from CometBFT protobuf types to ABCI protobuf types? Silly question but why does it need to exist? The types seem _almost_ interchangeable.
1. Was this broken upstream or a Celestia specific modification in celestia-core caused this issue? It looks like upstream is broken too: https://github.com/cometbft/cometbft/blob/e0e50aa9fca730ec208b67b54e48c8bdc630c17b/types/protobuf.go#L100-L109
1. Should we add a unit test in protobuf_test.go to prevent regressions?

Reply:
this isn't broken on main, where we just use the toProto method. here I wanted to minimize the number the changes, but the best approach would be to do what main does, which is to delete these and only use the ToProto methods and have round trip unit tests